.github/workflows/publish-release.yml

```diff
@@ -0,0 +1,114 @@
+name: Publish Release
+on: workflow_dispatch
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        ref: ${{ github.event.pull_request.head.sha }}
+
+    - name: Setup node version
+      uses: actions/setup-node@v2
+      with:
+        node-version: '14'
+        cache: 'npm'
+        cache-dependency-path: '**/package-lock.json'
+
+    - name: Setup NPM version
+      run: npm install -g npm@7
+
+    - name: Get NPM cache directory
+      id: npm-cache-dir
+      run: echo "::set-output name=dir::$(npm config get cache)"
+
+    - name: Restore NPM cache
+      id: npm-cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.npm-cache-dir.outputs.dir }}
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    - name: Install node modules
+      run: npm ci --ignore-scripts
+
+    - name: Download prebuild PR artifacts
+      uses: dawidd6/action-download-artifact@d0f291cf39bd21965ea9c4c6e210fc355c3844ed
+      with:
+        workflow: pr-realm-js.yml
+        commit: ${{ github.sha }}
+        path: ${{ github.workspace }}
+        workflow_conclusion: completed
+
+    - name: Read version
+      id: get-version
+      run: |
+        pkgVersion=$(npm pkg get version)
+        echo "::set-output name=version::$pkgVersion"
+      shell: bash
+
+    - name: Npm Publish realm-js-${{ steps.get-version.outputs.version }}
+      run: npm publish ${{ steps.get-version.outputs.version }}
+      env:
+        NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+    - name: Find Release PR
+      uses: juliangruber/find-pull-request-action@f9f7484f8237cf8485e5ab826e542ba5dd9e9c6e
+      id: find-pull-request
+      with:
+        branch: ${{ github.ref }}
+
+    - name: Merge Pull Request
+      uses: juliangruber/merge-pull-request-action@8a13f2645ad8b6ada32f829b2fae9c0955a5265d
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        number: ${{ steps.find-pull-request.outputs.number }}
+        method: squash
+
+    - name: Extract Changelog
+      run: scripts/extract-changelog.sh ${{ steps.get-version.outputs.version }} > ExtractedChangelog.md
+
+    - name: Create Github Release
+      uses: ncipollo/release-action@10c84d509b28aae3903113151bfd832314964f2e
+      with:
+        bodyFile: ExtractedChangelog.md
+        name: ${{ steps.get-version.outputs.version }}
+        tag: v${{ steps.get-version.outputs.version }}
+        token: ${{ secrets.GITHUB_TOKEN }}
+        draft: false
+
+    - name: Upload Github Release artifacts
+      run: npm prebuild --upload ${{ secrets.GITHUB_TOKEN }}
+
+    - name: 'Post to #realm-releases'
```

Comment:
That should probably be the last step in the workflow as it's not unlikely that it fails. The `release-to-slack` action will attempt to convert markdown to the slack post format, but that's not always flawless, especially when more complex markdown is used.

Reply:
Hi I'm assisting iamer on this PR.

Instead of moving the job around, would it be better to add `continue-on-error: true` here? That way the workflow retains its logical order, and also the whole release won't be marked as "failed" just because it wasn't posted on slack.

Reply:
I'm happy with `continue-on-error: true`